substratevm/src/com.oracle.svm.core.posix/src/com/oracle/svm/core/posix/PosixJavaNetSubstitutions.java

```diff
@@ -119,84 +419,1339 @@ protected int peekData(DatagramPacket p) throws IOException {
+            //   970      }
+        }
+
+        //   972      if ((fd = JVM_Socket(domain, SOCK_DGRAM, 0)) == JVM_IO_ERR) {
+        if ((fd = Socket.socket(domain, Socket.SOCK_DGRAM(), 0)) == Target_jvm.JVM_IO_ERR()) {
+            //   973          NET_ThrowByNameWithLastError(env, JNU_JAVANETPKG "SocketException",
+            //   974                         "Error creating socket");
+            throw new SocketException(PosixUtils.lastErrorString("Error creating socket") + " (Error creating socket)");
+            //   975          return;
+            /* Unreachable. */
+        }
+
+        //   978  #ifdef AF_INET6
+        if (IsDefined.socket_AF_INET6()) {
+            //   979      /* Disable IPV6_V6ONLY to ensure dual-socket support */
+            //   980      if (domain == AF_INET6) {
+            if (domain == Socket.AF_INET6()) {
+                //   981          arg = 0;
+                arg = 0;
+                CIntPointer arg_Pointer = StackValue.get(CIntPointer.class);
+                arg_Pointer.write(arg);
+                //   982          if (setsockopt(fd, IPPROTO_IPV6, IPV6_V6ONLY, (char*)&arg,
+                //   983                         sizeof(int)) < 0) {
+                if (Socket.setsockopt(fd, NetinetIn.IPPROTO_IPV6(), NetinetIn.IPV6_V6ONLY(), arg_Pointer, SizeOf.get(CIntPointer.class)) < 0) {
+                    //   984              NET_ThrowNew(env, errno, "cannot set IPPROTO_IPV6");
+                    try {
+                        JavaNetNetUtilMD.NET_ThrowNew(Errno.errno(), "cannot set IPPROTO_IPV6");
+                    } finally {
+                        //   985              close(fd);
+                        //   986              return;
+                        Unistd.close(fd);
+                        /* return required to prevent compiler from complaining about unhandled InterruptedIOException */
+                        return;
+                    }
+                }
+            }
+        }
+        //   987          }
+        //   988      }
+        //   989  #endif /* AF_INET6 */
+
+
+        //   991  #ifdef __APPLE__
+        if (IsDefined.__APPLE__()) {
+            //   992      arg = 65507;
+            arg = 65507;
+            CIntPointer arg_Pointer = StackValue.get(CIntPointer.class);
+            arg_Pointer.write(arg);
+            //   993      if (JVM_SetSockOpt(fd, SOL_SOCKET, SO_SNDBUF,
+            //   994                         (char *)&arg, sizeof(arg)) < 0) {
+            if (Socket.setsockopt(fd, Socket.SOL_SOCKET(), Socket.SO_SNDBUF(), arg_Pointer, SizeOf.get(CIntPointer.class)) < 0) {
+                try {
+                    //   995          getErrorString(errno, tmpbuf, sizeof(tmpbuf));
+                    //   996          JNU_ThrowByName(env, JNU_JAVANETPKG "SocketException", tmpbuf);
+                    JavaNetNetUtilMD.NET_ThrowNew(Errno.errno(), PosixUtils.lastErrorString("cannot set socket option"));
```

Comment:
Why aren't these (all) `throw new SocketException(PosixUtils.lastErrorString(PosixUtils.lastErrorString("cannot set socket option")))` like the C code says?  With the default string changed appropriately on the other instances?

Or be completely pedantic and after declaring `String tmpbuf` at the top of the method, say 
```
tmpbuf = PosixUtils.lastErrorString("cannot set socket option");
throw new SocketException(tmpbuf);
```